tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +59,43 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experimentName$ = this.getExperimentId$.pipe(
+    filter(Boolean),
+    mergeMap((experimentId) => {
+      return this.store.select(getExperiment, {experimentId});
+    }),
+    map((experiment) => (experiment ? experiment.name : null))
+  );
+
+  readonly title$ = this.store.select(getEnvironment).pipe(
+    combineLatestWith(this.experimentName$),
+    map(([env, experimentName]) => {
+      const tbBrandName = this.customBrandName
+        ? this.customBrandName
+        : 'TensorBoard';
```

Comment:
Since we're using this hardcoded string again on L93, could we extract it into something like `const DEFAULT_BRAND_NAME = 'TensorBoard'` near the top of the file?

optional: `this.customBrandName || DEFAULT_BRAND_NAME` also works and is shorter, if you prefer

Reply:
The concise version is much better, thanks! Also removed the magic constant `TensorBoard` as suggested.